report/src/main/kotlin/com/hartwig/actin/report/pdf/tables/trial/EligibleTrialGenerator.kt

```diff
@@ -85,124 +86,150 @@ class EligibleTrialGenerator(
 
     companion object {
 
-        fun nationalOpenCohorts(
+        fun localAndNationalExternalOpenAndEligibleCohorts(
             cohorts: List<InterpretedCohort>,
             externalTrials: ExternalTrials,
             requestingSource: TrialSource?,
             countryOfReference: Country?,
-            trialType: TrialType
+            localTrialsType: LocalTrialsType
         ): TrialTableGenerator {
             val nationalExternalTrials = ExternalTrialSummarizer.summarize(externalTrials.nationalTrials.filtered)
             val nationalExternalTrialFilteredCount = ExternalTrialSummarizer.summarize(externalTrials.excludedNationalTrials()).size
 
-            return forOpenCohorts(
-                cohorts = cohorts,
-                externalTrials = nationalExternalTrials,
-                externalFilteredCount = nationalExternalTrialFilteredCount,
+            return forLocalAndNationalExternalOpenAndEligibleLocalCohorts(
+                openAndEligibleLocalCohorts = filterOpenAndEligibleCohorts(localTrialsType, cohorts),
+                relevantExternalTrials = nationalExternalTrials,
+                relevantExternalTrialsFilteredCount = nationalExternalTrialFilteredCount,
                 requestingSource = requestingSource,
                 countryOfReference = countryOfReference,
-                type = trialType
+                trialDescriptionString = trialDescriptionString(localTrialsType, countryOfReference),
             )
         }
 
-        fun internationalExternalOpenCohorts(
+        fun externalOpenAndEligibleCohorts(
             externalTrials: ExternalTrials,
-            requestingSource: TrialSource?
+            requestingSource: TrialSource?,
+            isNational: Boolean
         ): TrialTableGenerator {
-            val internationalExternalTrials = ExternalTrialSummarizer.summarize(externalTrials.internationalTrials.filtered)
-            val internationalExternalTrialFilteredCount = ExternalTrialSummarizer.summarize(externalTrials.excludedInternationalTrials()).size
+            val (relevantExternalTrials, relevantExternalTrialsFilteredCount) =
+                if (isNational) {
+                    ExternalTrialSummarizer.summarize(externalTrials.nationalTrials.filtered) to
+                            ExternalTrialSummarizer.summarize(externalTrials.excludedNationalTrials()).size
+                } else {
+                    ExternalTrialSummarizer.summarize(externalTrials.internationalTrials.filtered) to
+                            ExternalTrialSummarizer.summarize(externalTrials.excludedInternationalTrials()).size
+                }
```

Comment:
You could factor out the summarization:
```suggestion
            val (includedTrials, excludedTrials) = if (isNational) {
                externalTrials.nationalTrials.filtered to externalTrials.excludedNationalTrials()
            } else {
                externalTrials.internationalTrials.filtered to externalTrials.excludedInternationalTrials()
            }
            val relevantExternalTrials = ExternalTrialSummarizer.summarize(includedTrials)
            val relevantExternalTrialsFilteredCount = ExternalTrialSummarizer.summarize(excludedTrials).size
```

Reply:
nice! cleaner